ui/apps/platform/src/Containers/NetworkGraph/deployment/ExternalFlows.tsx

```diff
@@ -53,11 +45,15 @@ function ExternalFlows({ deploymentId }: InternalFlowsProps) {
         setInitialCIDRFormValue(tempFilter.externalIP);
     };
 
-    // TODO: Show all unique ports
-    const allUniquePorts = getAllUniquePorts([]);
     const isFiltered = appliedFilter.externalIP !== '';
 
-    // TODO: Filter network flows based on the match type and external IP
+    const advancedFilters = useMemo(
+        () => ({
+            'Deployment ID': deploymentId,
+            'External Source Address': appliedFilter.externalIP,
+        }),
+        [appliedFilter.externalIP, deploymentId]
+    );
```

Comment:
This keeps the same reference between renders, preventing unnecessary updates. We get wonky behavior otherwise.